### ISSUE TITLE
The TensorFlow library was compiled to use SSE4.1 instructions, but these aren't available on your machine

### ISSUE THREAD

**Issue by Ratione** [openai/spinningup#86]
Install Problem here:

I am running an Ubuntu 18.04 VM, on a windows host machine. I got through the install with minimal problems, and compiled Tensorflow as everyone suggested. I am not totally sure what information is even relevant here, but ultimately, I cannot get past the "Check your Install" section. I run 

`python -m spinup.run ppo --hid "[32,32]" --env LunarLander-v2 --exp_name installtest --gamma 0.999`

and I end up getting this as an error message. 

'2018-12-29 09:09:02.787690: F tensorflow/core/platform/cpu_feature_guard.cc:37] The TensorFlow library was compiled to use SSE4.1 instructions, but these aren't available on your machine.
Aborted (core dumped)'

Nothing I've found on SO or various other sites has helped. Any advice or guidance you could provide would be greatly appreciated.

**Comment by ibarrond** [comment]
Same issue, same OS and environment, also in a VM. Please help!

**Comment by ibarrond** [comment]
My solution was quite particular: I found out that I could change the type of CPU in the configuration of my VM, thus I assigned a nore modern CPU with SSE capabilities included. Solved!

**Comment by Ratione** [comment]
Currently, I am experimenting with various settings in my VM, Ill update later today if I get it working or run into any walls.

**Comment by Ratione** [comment]
Okay, so today is tomorrow, and tomorrow is actually Sunday, but nevertheless I got it working.

I started with a fresh install of Ubuntu 18.04 LTS, on VMware player (the free version), and made sure that virtualization was enabled in my BIOS. Standard install, Launch, update system files, reboot, ensure everything is working properly, then go to the Anaconda website and download anaconda for Py3.7. After that, the order of operations was,

`conda create -n spinningup python=3.6`
`source activate spinningup`
`sudo apt-get update && sudo apt-get install libopenmpi-dev`
`conda install tensorflow`
,etc.

Very similar to the guide, but the 2 steps that were important were BOTH enabling virtualization in the Bios, and installing tensorflow with conda. I ran the test, and it ran for like 10m, just as the install guide said it would, and then successfully ran the simulation video with the given instructions. That was all I did, from fresh install to getting it running. It did however take me 3 days to figure out.

That info needs to be in the guide. The guide is for people who have _never_ done this before. As far as I can tell, that's the _whole_ point. Some people are going to look at the guide as it stands, and not be able to get it working, or figure out what the right question is to even ask, and those people are more than likely going to give up, in the guides current form. Please update the guide with these instructions, or whatever the "Right" instructions are, for the _whole_ install process. That _has_ to include tensorflow.

**Comment by Ratione** [comment]
Not necessarily. So far, the Conda install has been enough for everything I've tried. But there may be some configuration of TensorFlow that is optimal for spinning up, or it may be that the method I used _doesn't_ actually work with all the examples. If it is _not_ the proper way to do things, the correct way needed to be detailed.

Forgive me for this massive wall of text, but I feel like this is a point worth proving. 

The problem is that the guide only mentions that TensorFlow is required in one place, right now, and that's not even in the actual **instructions** part of the install instructions. There is even a section that says: 

> Spinning Up requires Python3, OpenAI Gym, and OpenMPI

With no mention of TensorFlow. I understood that it required TensorFlow when I read it, Spinning Up does mention it in other places, but for the absolute beginner, I totally understand why this is so intimidating. 

It does mention that there is a pip install, but the pip install is what was giving me an the error in the first place. It was a different error, than the one I listed, but that's not exactly my point. I understand, for the most part, what was causing the issue now, and learned a valuable lesson about development environments and the whole "It runs on my system" thing, but that's not really my point either.

My point is; Spinning Up is aimed at people with no background in AI, it says that in the introduction. What I was struggling with is **not** an edge case. I had worked with tensorflow before, and had been _explicitly_ trying to find a guide to the setup process for weeks before I this, so I could make a bare-bones install to try to lower the barrier for some of my non-programmer friends. If the guide gets more popular than it is already, which it appears to be doing, it is quickly going to hit a threshold where people are trying to follow it with _no_ programming experience, and it is advertising itself as being more accessible.

When I was trying to diagnose the problem, I found dozens of posts about people struggling to install TensorFlow on their systems on their system. Adding a conda environment makes that process _more_ complicated, not less, if you aren't already familiar with how python environments work. The guide already assumes that that might be the case. 

I do not know enough about the subject to say what the right instructions are for the general case. Maybe the conda install is fine, and all the guide needs is another line for that, If that is not the right way however, then the guide needs to detail whatever that way is, in however much detail is necessary to be generally applicable _for SpinningUp_, or at least link to somewhere that does. The guide does something similar to that for python environments already, but more to the point, I did not find _anywhere_ that went through the install process, or even clarified who the various 20-some-odd install options were actually for, in a language that you could understand without a background AI, and at least a bachelors in CS. Luckily for me, I am fortunate enough to have the latter. 

The guide isn't bad at all, once you get past that first step. Not by any means. It is probably the most concise and accessible guide I have come across, but it still suffers from the same problem that every guide I've looked at suffers from, albeit to a lesser degree; It makes too many assumptions about what the reader already knows. The only exception to that I have found has been the DLB, and calling that thing concise would be calling a mountain a molehill. "Exhaustive", might be a better description.

If the whole, `conda install tensorflow`, thing is not the right way to install it for whatever reason, then yeah, you or someone who works on the project and understands these things probably need to write a section on installing tensorflow. It was my understanding that that was the whole point: 

> "[Our] package here is designed to serve as the missing middle step for people who are excited by deep RL, and would like to learn how to use it or make a contribution, but don’t have a clear sense of what to study or how to transmute algorithms into code. We’ve tried to make this as helpful a launching point as possible.

>That said, practitioners aren’t the only people who can (or should) benefit from these materials. Solving AI safety will require people with a wide range of expertise and perspectives, and many relevant professions have no connection to engineering or computer science at all..."

I would love to contribute. I will when I can, but right now I am still an amateur at all this, and I have been lucky to get this far without a formal training in the subject.

**Comment by programmin1** [comment]
I get this error and I am not in a VM. With:
AMD A6-3620 APU with Radeon(tm) HD Graphics

**Comment by ghost** [comment]
I have a AMD a4-3400 APU with Radeom(tm) HD Graphics Too and i face the same error

**Comment by sebasmos** [comment]
I am using an old AMD, so this error still exist there. Commenting out this [line](https://github.com/tensorflow/tensorflow/blob/c4b09b5df79625a70853fd66b5caa7dd92fb4d1f/tensorflow/tensorflow.bzl#L130) worked for you? For which TF versions (?)
